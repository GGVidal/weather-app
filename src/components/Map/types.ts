import { LatLng } from "react-native-maps";

export interface MapProps {
  latitude: number;
  longitude: number;
}
