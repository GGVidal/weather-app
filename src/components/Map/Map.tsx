import React, { FC } from "react";
import { StyleSheet, Dimensions } from "react-native";
import MapView, { Marker } from "react-native-maps";
import { MapProps } from "./types";

const DELTA_VALUE = 0.1;

export const Map: FC<MapProps> = ({ latitude, longitude }) => {
  const coordObj = {
    latitude,
    longitude,
  };
  return (
    <MapView
      region={{
        ...coordObj,
        latitudeDelta: DELTA_VALUE,
        longitudeDelta: DELTA_VALUE,
      }}
      style={styles.map}
    >
      <Marker
        title="My Location"
        description="My location"
        coordinate={coordObj}
      />
    </MapView>
  );
};

const styles = StyleSheet.create({
  map: {
    width: Dimensions.get("window").width,
    height: Dimensions.get("window").height / 2,
  },
});
