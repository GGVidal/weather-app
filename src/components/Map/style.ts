import MapView from "react-native-maps";
import styled from "styled-components/native";
import { Dimensions } from "react-native";

export const MapContainer = styled(MapView)`
  width: ${Dimensions.get("window").width - 30}px;
  height: ${Dimensions.get("window").height / 2}px;
  margin-top: 20px;
  margin-bottom: 40px;
  border-radius: 10px;
`;
