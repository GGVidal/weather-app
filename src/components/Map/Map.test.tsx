import React from "react";
import TestRenderer from "react-test-renderer";

import { Map } from "../Map";

describe("renders correctly", () => {
  it("works", () => {
    const renderer = TestRenderer.create(
      <Map
        latitude={0}
        longitude={0}
        address={{
          city: null,
          district: null,
          streetNumber: null,
          street: null,
          region: null,
          subregion: null,
          country: null,
          postalCode: null,
          name: null,
          isoCountryCode: null,
          timezone: null,
        }}
      />
    );
    expect(renderer).toMatchSnapshot();
  });
});
