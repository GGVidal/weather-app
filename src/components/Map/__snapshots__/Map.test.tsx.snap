// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly works 1`] = `
<AIRMap
  initialRegion={null}
  onChange={[Function]}
  onLayout={[Function]}
  onMapReady={[Function]}
  onMarkerPress={[Function]}
  region={
    Object {
      "city": "",
      "district": "",
      "latitude": 0,
      "latitudeDelta": 0.1,
      "longitude": 0,
      "longitudeDelta": 0.1,
      "street": "",
      "streetNumber": "",
    }
  }
  scrollDuringRotateOrZoomEnabled={true}
  style={
    Array [
      Object {
        "borderRadius": 10,
        "height": 667,
        "marginBottom": 40,
        "marginTop": 20,
        "width": 720,
      },
    ]
  }
  zoomControlEnabled={true}
  zoomTapEnabled={true}
>
  <AIRMapMarker
    coordinate={
      Object {
        "city": "",
        "district": "",
        "latitude": 0,
        "longitude": 0,
        "street": "",
        "streetNumber": "",
      }
    }
    description="null, null"
    onPress={[Function]}
    stopPropagation={false}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "position": "absolute",
        },
        undefined,
      ]
    }
    title="null, null"
  />
</AIRMap>
`;
